rl4co/models/zoo/am/policy.py

```diff
@@ -17,18 +26,68 @@ class AttentionModelPolicy(AutoregressivePolicy):
 
     def __init__(
         self,
-        env_name: str,
-        embedding_dim: int = 128,
+        encoder: nn.Module = None,
+        decoder: nn.Module = None,
```

Comment:
In this case we should not allow `None` neither for the encoder, nor the decoder. The reason being that for the Attention Model, we need to pre-compute some embeddings taken from the encoder.
If one wants to modify this could just pass `NoEncoder` with a modified Decoder that does not need to precompute the cache